presto-main/src/main/java/com/facebook/presto/block/BlockUtils.java

```diff
@@ -52,7 +52,7 @@ else if (value instanceof String) {
             type.writeSlice(blockBuilder, slice, 0, slice.length());
         }
         else {
-            throw new IllegalArgumentException("Unsupported type: " + value.getClass().getName());
```

Comment:
If all primitive types is excluded here, what about `Integer`? I found it weird either way.